storagemarket/impl/client.go

```diff
@@ -184,12 +187,53 @@ func (c *Client) ListProviders(ctx context.Context) (<-chan storagemarket.Storag
 }
 
 // ListLocalDeals lists deals initiated by this storage client
-func (c *Client) ListLocalDeals(ctx context.Context) ([]storagemarket.ClientDeal, error) {
+func (c *Client) ListLocalDeals(ctx context.Context, params ...storagemarket.ListDealsPageParams) ([]storagemarket.ClientDeal, error) {
+	if len(params) > 1 {
+		return nil, errors.New("cannot pass more than one filter param")
+	}
+
+	// fetch deals from the state machine
 	var out []storagemarket.ClientDeal
 	if err := c.statemachines.List(&out); err != nil {
 		return nil, err
 	}
-	return out, nil
+	// return all deals if NO filter has been passed.
+	if len(params) == 0 {
+		return out, nil
+	}
+
+	filter := params[0]
+	// If NDeals isn't valid, show ALL deals that meet the criteria.
+	if filter.DealsPerPage <= 0 {
+		filter.DealsPerPage = math.MaxInt32
+	}
+	// MaxEndEpoch is a noop if it hasn't been configured.
+	if filter.MaxEndEpoch <= 0 {
+		filter.MaxEndEpoch = math.MaxInt64
+	}
+
+	// filter and return
+	filtered := make([]storagemarket.ClientDeal, 0, len(out))
```

Comment:
If DealsPerPage is specified, let's initialize this array with a capacity of DealsPerPage

Reply:
Done.